### PR TITLE
feat: now you can share the StarSearch waitlist on socials

### LINKED ISSUE #3283: Feature: Add a share to socials buttons on StarSearch waitlist page
As a potential user of StarSearch, I want to have social share buttons on the waitlist page after I sign up so that I can share it with others.

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

This PR adds social share buttons for X/Twitter and LinkedIn

## Related Tickets & Documents

Closes #3283

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/833231/5d16fe4f-0789-4ea4-ade8-bced4dcc4d32

Twitter share

CleanShot 2024-04-30 at 11 28 40

LinkedIn share

CleanShot 2024-04-30 at 11 29 16

Note that on LinkedIn, the OG image won't be that big once it's in a post.

## Steps to QA

Note, only the core team can really QA this as they need to reset their user profile in the development database.

To reset being on the waitlist, you'll need to run the following:

```SQL
UPDATE
	users
SET
	is_waitlisted = FALSE
WHERE
	ID = 833231; # put your user ID here instead

SELECT
	is_waitlisted,
	id
FROM
	users
WHERE
	ID = 833231; # put your user ID here instead
```

Core team members will also need to ensure these properties are false (temporarily) in Posthog. You can ask @isabensusan or myself to change it temporarily if you do not have access.

CleanShot 2024-04-24 at 09 04 16

**Logged out (not on waitlist)**

1. Go to `/star-search`
2. You're redirected to `/star-search/waitlist`.
3. Click on the `Sign up to Join the Waitlist` button.
4. You're logged in and automatically added to the waiting list.

**Logged out (on waitlist)**

1. Go to `/star-search`
2. You're redirected to `/star-search/waitlist`.
3. Click on the `Sign up to Join the Waitlist` button.
4. You're logged in and a message says you're already on the waitlist.

**Logged in (not on waitlist)**

1. Go to `/star-search`
2. You're redirected to `/star-search/waitlist`.
3. Click on the `Join the Waitlist` button.
4. You're added to the waiting list.

**Logged in (on waitlist)**

1. Go to `/star-search`
2. You're redirected to `/star-search/waitlist`.
3. A message says you're already on the waitlist.

**Use the social share buttons**

In all the test scenarios above, you'll end up with two social share buttons. One for X/Twitter and one for LinkedIn. For LinkedIn, all you can do is share a URL, no text.

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

### CODE REVIEW

File: pages/star-search/waitlist.tsx [hunk 1]
```diff
@@ -155,21 +191,36 @@ export default function StarSearchWaitListPage({
...
-                    You&apos;re already on the waitlist along with {humanizeNumber(waitlistCount)} other people.
-                  </p>
-                ) : (
-                  <p className="grid place-content-center">
-                    You&apos;re in along with {humanizeNumber(waitlistCount)} other people on the StarSearch waitlist!
-                  </p>
-                )}
-              </>
+              <div className="flex gap-4">
+                <Button
```
Comment (nickytonline) [inline pages/star-search/waitlist.tsx:205]: If we plan to do other social share buttons for these platform in the future, we can create reusuable social share buttons.

File: pages/star-search/waitlist.tsx [hunk 2]
```diff
@@ -139,6 +162,19 @@ export default function StarSearchWaitListPage({
...
   }
 
+  const tweetQueryParams = new URLSearchParams();
+  const linkedInQueryParams = new URLSearchParams();
+
+  if (isWaitlisted) {
+    const url = new URL("/star-search/waitlist", process.env.NEXT_PUBLIC_BASE_URL || "http://localhost:3000");
+    tweetQueryParams.set(
+      "text",
+      `I just joined the waitlist for @saucedopen's StarSearch, a tool that provides AI-powered insights based on contributor data. Join me! ${url}`
```
Comment (nickytonline) [inline pages/star-search/waitlist.tsx:172]: Let me know if this wording is OK @bdougie.

Comment (bdougie) [inline pages/star-search/waitlist.tsx:172]:
```suggestion
      `I just joined the waitlist for @saucedopen's StarSearch, Copilot for git history. Join me! ${url}`
```

 per our slack thread. Use this copy.

**Review by bdougie — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

File: pages/star-search/waitlist.tsx [hunk 3]
```diff
@@ -148,28 +184,43 @@ export default function StarSearchWaitListPage({
...
           </div>
-          <p className="font-semibold text-5xl text-center tracking-tight text-balance">Copilot, but for git history</p>
-          <p className="text-center">Ask anything, get AI powered insights based on contributor data</p>
+          <p className="font-semibold text-5xl text-center tracking-tight text-balance">
+            {getTitle({ sessionToken, isWaitlisted, waitlistCount, alreadyOnWaitlist })}
+          </p>
+          <p className="text-center">
+            {isWaitlisted
+              ? "Share with others to get early access sooner"
+              : "Ask anything, get AI powered insights based on contributor data"}
```
Comment (brandonroberts) [inline pages/star-search/waitlist.tsx:200]: nit: more of a question with naming?

```suggestion
              : "Ask anything, get AI-powered insights based on contributor data"}
```

**Pull request merged**